### DISCUSSION TITLE
[Question]: What would be needed to add support for the code interpreter showing plotly plots?

### DISCUSSION THREAD

**Discussion by adrianfagerland** [danny-avila/LibreChat#4367] [Q&A]
### What is your question?

I need to create an assistant that can display plotly plots inline in the chat, and I will implement this, and create a PR when done. But before I do that, I want to know if this is something that is desirable in the first place.

Also, it seems the code interpreter, at least for me, has a bug in which the text in the markdown cell gets displayed as duplicated in a weird manner. I'm not sure if you're aware of this but I could create an issue with more details if wanted.

### More Details

I have looked through the code of the code interpreter but it seems that there is a some clutter that has gathered in that part of the code, so I need to spend more time to get a grasp of it.

### What is the main subject of your question?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I’m not sure what you’re referring to, some screenshots might help me understand

**Reply by adrianfagerland** [reply]
![image](https://github.com/user-attachments/assets/eda39e2b-c8c2-4831-bb39-074aaa662f3f)

Here. Usually it disappears when it is done with writing the code inside it, but with this one it stayed like this

**Reply by danny-avila** [reply]
I see. This looks like a specific issue with Assistants where the tool calls in progress are being accumulated incorrectly, but this shouldn't affect whether the plots show or not. That would be a limitation on OpenAI's side.
